Code/Source/Wrappers/RglTexture.cpp

```diff
@@ -0,0 +1,216 @@
+        {
+            const auto albedoTexImagePropVal = propertyValues.at(albedoTexPropIdx.GetIndex());
+            const auto& imageAsset = albedoTexImagePropVal.GetValue<AZ::Data::Asset<AZ::RPI::ImageAsset>>();
+            imageRglTexture = CreateFromImageAsset(imageAsset.GetId());
+        }
+
+        if (imageRglTexture.IsValid())
+        {
+            return imageRglTexture;
+        }
+
+        if (const auto albedoColorPropIdx = propLayout->FindPropertyIndex(albedoColorName); !albedoColorPropIdx.IsNull())
+        {
+            imageRglTexture =
+                AZStd::move(CreateFromFactor(CreateGrayFromColor(propertyValues.at(albedoColorPropIdx.GetIndex()).GetValue<AZ::Color>())));
+        }
+        else
+        {
+            AZ_Error(
+                TraceWindowName.c_str(),
+                false,
+                "Unable to find specular color and texture properties of material asset with ID: %s.",
+                id.ToString<AZStd::string>().c_str());
+        }
+
+        return imageRglTexture;
+    }
+
+    RglTexture RglTexture::CreateFromFactor(float factor)
+    {
+        auto factor8 = aznumeric_cast<AZ::u8>(factor * 255.0f);
+        return RglTexture{ &factor8, 1, 1 };
+    }
+
+    float RglTexture::CreateGrayFromColor(const AZ::Color& color)
+    {
+        return RedGrayMultiplier * color.GetR() + GreenGrayMultiplier * color.GetG() + BlueGrayMultiplier * color.GetB();
+    }
+
+    uint8_t RglTexture::CreateGray8FromColor(const AZ::Color& color)
+    {
+        return static_cast<uint8_t>(CreateGrayFromColor(color) * 255.0f);
+    }
+
+    RglTexture RglTexture::CreateFromImageAsset(const AZ::Data::AssetId& imageAssetId)
+    {
+        // Made a static member to minimize reallocations, since multiple assets will be processed.
+        static AZStd::vector<uint8_t> tempRglTextureData;
+
+        RglTexture imageRglTexture = CreateInvalid();
+
+        if (!imageAssetId.IsValid())
+        {
+            return imageRglTexture;
+        }
+
+        AZ::Data::Asset<AZ::RPI::StreamingImageAsset> imageAsset =
+            AZ::Data::AssetManager::Instance().GetAsset<AZ::RPI::StreamingImageAsset>(
+                imageAssetId,
+                AZ::Data::AssetLoadBehavior::PreLoad,
+                AZ::Data::AssetLoadParameters(nullptr, AZ::Data::AssetDependencyLoadRules::LoadAll));
+        imageAsset.QueueLoad();
+        imageAsset.BlockUntilLoadComplete();
+
+        const AZ::RHI::ImageDescriptor imageDescriptor = imageAsset->GetImageDescriptor();
+
+        using Format = AZ::RHI::Format;
+        const auto format = imageDescriptor.m_format;
+        const auto& size = imageDescriptor.m_size;
+        if (format == Format::BC1_UNORM || format == Format::BC1_UNORM_SRGB || format == Format::BC4_UNORM)
+        {
+            tempRglTextureData.clear();
+            tempRglTextureData.resize(size.m_width * size.m_height);
```

Comment:
```suggestion
            tempRglTextureData.resize(size.m_width * size.m_height);
```
It seems that you should not need to .clear at all. You resize the texture data to its target dimensions and then set each pixel's value without any mask or filtering.